tasks/nc_apps.yml

```diff
@@ -2,59 +2,46 @@
 - name: "[App] - Parse the item values"
   ansible.builtin.set_fact:
     nc_app_name: "{{ item.key }}"
+    nc_app_source: "{{ item.value.source if item.value.source is defined else item.value }}"
     nc_app_cfg: "{{ item.value }}"
 
-- name: "Install and enable Nextcloud Apps"
-  when: nc_app_cfg is string
+- name: "[App] - Download, install, enable and if provided configure Nextcloud Apps"
   block:
-    - name: "[App] - Download Archive in apps folder {{ nc_app_name }}"
-      ansible.builtin.unarchive:
-        copy: false
-        src: "{{ nc_app_cfg }}"
-        dest: "{{ nextcloud_webroot }}/apps/"
-        owner: "{{ nextcloud_websrv_user }}"
-        group: "{{ nextcloud_websrv_group }}"
-        creates: "{{ nextcloud_webroot }}/apps/{{ nc_app_name }}"
-      when: nc_app_cfg is not none
+    - name: "[App] - Verify the yaml declaration for the app \"{{ nc_app_name }}\""
+      ansible.builtin.assert:
+        that:
+          - (nc_app_cfg.source is defined) and (nc_app_cfg.source is string)
+        msg: "{{ nc_app_name }} is not well declared."
+      when: nc_app_cfg is not string
 
-    - name: "[App] - Enable the application {{ nc_app_name }}"
+    - name: "[App] - Install the app \"{{ nc_app_name }}\" from the official app-store"
```

Comment:
I'm not sure if `\"` syntax would be rendered properly.
Leaving this comment as a reminder to check Molecule result after run.

Reply:
Is there anything I need to do?

Reply:
Not for now. I've just left this comment for future check when Molecule will run.